packages/@dcl/inspector/src/components/Block/Block.tsx

```diff
@@ -1,12 +1,40 @@
-import React from 'react'
+import React, { useEffect } from 'react'
+import { useDrop } from 'react-dnd'
+
 import { Props } from './types'
+
 import './Block.css'
 
-const Block: React.FC<React.PropsWithChildren<Props>> = (props) => {
+const Block: React.FC<React.PropsWithChildren<Props>> = ({
+  acceptDropTypes = [],
+  label,
+  children,
+  onDrop  = () => null,
+  onDropHover = () => null
+}) => {
+  const [{ isActive }, drop] = useDrop(
+    () => ({
+      accept: acceptDropTypes,
+      drop: (val, monitor) => {
+        if (monitor.didDrop()) return
+        onDrop(val)
+      },
+      collect: (monitor) => ({
+        isActive: monitor.canDrop() && monitor.isOver(),
+      })
+    }),
+    [label, children]
+  )
+
+  useEffect(() => {
+    onDropHover(isActive)
+    return () => onDropHover(false)
+  }, [isActive])
+
   return (
-    <div className="Block">
-      {props.label && <label>{props.label}</label>}
-      <div className="content">{props.children}</div>
+    <div ref={drop} className="Block">
```

Comment:
I would suggest, instead of adding the drop functionality into the `Block` component, which ends up with logic split between this component and the parent, and props being passed around, to just foward the reference in the `Block` component and let the parent use the ref for any use case that would require a `ref`. 

For example: 

```tsx
const Block = React.fowardRef((props, ref) => {
  // ...
  return (
    <div className="Block" ref={ref}>
      {/* ... */}
    </div>
  )
})
```

The the parent of `Block` can do do the `useDrop` or any other hook and pass the ref to like `<Block ref={drop}>`